tests/lib/rules/no-unused-prop-types.js

```diff
@@ -2111,6 +2111,24 @@ ruleTester.run('no-unused-prop-types', rule, {
       parser: 'babel-eslint',
       options: [{skipShapeProps: false}]
     }, {
+      // issue #1506
+      code: [
+        'class MyComponent extends React.Component {',
+        '  onFoo() {',
+        '    this.setState((prevState, props) => {',
+        '      props.doSomething();',
```

Comment:
does this work independent of what "props" is called?

Reply:
Not yet, working a better version that will handle more cases. This code doesn't ] handle destructuring or anything like that.

Reply:
I think it's important that it handles that from the get-go.